### PR TITLE
chore: Organization seed

### PR DESCRIPTION
**Pull request by leog**
## What does this PR do?

Creates an `acme` organization with 10 member `marketing` sub-team, 15 member `sales` sub-team and 20 member `engineering` sub-team. All with random users. Introduces a few functions to be able to easily create more random fake users and sub-teams.

## Type of change

  - [X] Chore (refactoring code, technical debt, workflow improvements)

## How should this be tested?

Run the DB seeded and go to CAL_URL/acme

## Mandatory Tasks

  - [ ] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: packages/lib/package.json [hunk 1]
```diff
@@ -32,7 +32,7 @@
   "devDependencies": {
     "@calcom/tsconfig": "*",
     "@calcom/types": "*",
-    "@faker-js/faker": "^7.3.0",
+    "@faker-js/faker": "^8.0.2",
```
Comment (leog) [inline packages/lib/package.json:35]: TO be able to use `multiple` helper

File: packages/prisma/package.json [hunk 2]
```diff
@@ -20,6 +20,7 @@
     "seed-insights": "ts-node --transpile-only ./seed-insights.ts"
   },
   "devDependencies": {
+    "@faker-js/faker": "^8.0.2",
```
Comment (leog) [inline packages/prisma/package.json:23]: Added to prisma package be able to use in seeder

File: packages/prisma/seed.ts [hunk 3]
```diff
@@ -23,7 +25,7 @@ async function createUserAndEventType(opts: {
     role?: UserPermissionRole;
   };
   eventTypes: Array<
-    Prisma.EventTypeCreateInput & {
+    Prisma.EventTypeUncheckedCreateInput & {
```
Comment (leog) [inline packages/prisma/seed.ts:28]: More accurate type to avoid type errors when passing unchecked data

File: packages/prisma/seed.ts [hunk 4]
```diff
@@ -169,6 +171,51 @@ async function createTeamAndAddUsers(
...
+        ).map((result, idx, arr) => ({
+          role: arr.length - 1 === idx ? MembershipRole.ADMIN : MembershipRole.MEMBER,
+          accepted: true,
+          userId: result.id,
+        })),
+      },
+    }))
+  );
+}
+
```
Comment (leog) [inline packages/prisma/seed.ts:218]: To be able to generate a team with given name and member count

File: packages/prisma/seed.ts [hunk 5]
```diff
@@ -277,19 +324,19 @@ async function main() {
         title: "Zoom Event",
         slug: "zoom",
         length: 60,
-        locations: [{ type: zoomMeta.appData?.location.type }],
+        locations: [{ type: zoomMeta.appData?.location?.type }],
```
Comment (leog) [inline packages/prisma/seed.ts:327]: Type error fix

File: packages/prisma/seed.ts [hunk 6]
```diff
@@ -277,19 +324,19 @@ async function main() {
...
       },
       {
         title: "Daily Event",
         slug: "daily",
         length: 60,
-        locations: [{ type: dailyMeta.appData?.location.type }],
+        locations: [{ type: dailyMeta.appData?.location?.type }],
```
Comment (leog) [inline packages/prisma/seed.ts:333]: Type error fix

File: packages/prisma/seed.ts [hunk 7]
```diff
@@ -277,19 +324,19 @@ async function main() {
...
       },
       {
         title: "Google Meet",
         slug: "google-meet",
         length: 60,
-        locations: [{ type: googleMeetMeta.appData?.location.type }],
+        locations: [{ type: googleMeetMeta.appData?.location?.type }],
```
Comment (leog) [inline packages/prisma/seed.ts:339]: Type error fix

File: yarn.lock [hunk 8]
```diff
@@ -4288,7 +4288,7 @@ __metadata:
     "@calcom/dayjs": "*"
     "@calcom/tsconfig": "*"
     "@calcom/types": "*"
-    "@faker-js/faker": ^7.3.0
+    "@faker-js/faker": ^8.0.2
```
Comment (leog) [inline yarn.lock:4291]: Should I just keep `fakerjs` changes in `yarn.lock`?

File: packages/prisma/seed.ts [hunk 9]
```diff
@@ -604,6 +651,55 @@ async function main() {
       },
     ]
   );
+
+  // Org Owner
+  const orgUser1 = await createUserAndEventType({
```
Comment (alannnc) [inline packages/prisma/seed.ts:656]:
```suggestion
  const orgUserOwner = await createUserAndEventType({
```
nit

File: packages/prisma/seed.ts [hunk 10]
```diff
@@ -604,6 +651,55 @@ async function main() {
...
+      email: "john@acme.com",
+      password: "john",
+      username: "john",
+      name: "John Acme Org Owner",
+    },
+    eventTypes: [],
+  });
+
+  // Org Admin
+  const orgUser2 = await createUserAndEventType({
```
Comment (alannnc) [inline packages/prisma/seed.ts:667]:
```suggestion
  const orgUserAdmin = await createUserAndEventType({
```
nit

File: packages/prisma/seed.ts [hunk 11]
```diff
@@ -169,6 +171,51 @@ async function createTeamAndAddUsers(
   }
 }
 
+// Generates defined number of random users, with extraFields as needed
+async function generateFakeUsers({
+  count = 1,
+  extraFields,
+}: {
+  count: number;
+  extraFields?: Record<string, unknown>;
```
Comment (sean-brydon) [inline packages/prisma/seed.ts:180]: We can add the extact the type for createUserAndEventType and pick so we have some type saftey here 

```suggestion
    extraFields?: Pick<SeedUser, "timeZone" | "role">;
```

**Comment by leog** [comment]
Closing this one for now, not sure we need it.

**Pull request closed without merging**